### DISCUSSION TITLE
wrongly reports missing parameter for pydantic_settings's BaseSettings

### DISCUSSION THREAD

**Discussion by braindevices** [microsoft/pyright#8917] [Q&A]
Is this a bug?

When using pydantic Settings the member without default is valid and can be set by other config source.
But the pyright report `error: Argument missing for parameter "b" (reportCallIssue)`
Same code can pass mypy without issue.

**Code or Screenshots**
```python
from pydantic_settings import BaseSettings, SettingsConfigDict
class Settings3(BaseSettings):
    model_config = SettingsConfigDict(cli_parse_args=True)
    foobar: str = "default"
    b: int # required

Settings3() # this line cause problem, pyright expect the b, but in basesetting this is valid.
```

**VS Code extension or command-line**
command-line tool pyright 1.1.379

**Comment by erictraut** [comment]
Pyright is a standards-compliant type checker, and it has no special knowledge of pydantic or other libraries outside of the stdlib. It bases its type analysis on the type information provided by libraries. Pydantic uses the [`@dataclass_transform`](https://typing.readthedocs.io/en/latest/spec/dataclasses.html#the-dataclass-transform-decorator) decorator to indicate to type checkers that it behaves similar to the stdlib `@dataclass` decorator. In places where it deviates from dataclass, pyright won't have knowledge of this non-standard behavior.

Are you sure that the above code works? When I run it, I receive an exception on the same line where pyright reports an error — and for the same apparent reason that pyright is indicating.
